src/main/java/gyro/google/compute/RouterBgp.java

```diff
@@ -0,0 +1,154 @@
+    }
+
+    public void setAsn(Long asn) {
+        this.asn = asn;
+    }
+
+    /**
+     * The mode to use for advertisement. Valid values are ``DEFAULT`` or ``CUSTOM``.
+     */
+    @ValidStrings({ "DEFAULT", "CUSTOM" })
+    @Updatable
+    public String getAdvertiseMode() {
+        return advertiseMode;
+    }
+
+    public void setAdvertiseMode(String advertiseMode) {
+        this.advertiseMode = advertiseMode;
+    }
+
+    /**
+     * The list of prefix groups to advertise in ``CUSTOM`` mode. Valid values are ``ALL_SUBNETS`` or ``ALL_VPC_SUBNETS`` or ``ALL_PEER_VPC_SUBNETS``.
+     */
+    @ValidStrings({ "ALL_SUBNETS", "ALL_VPC_SUBNETS", "ALL_PEER_VPC_SUBNETS" })
+    @Updatable
+    public List<String> getAdvertisedGroups() {
+        if (advertisedGroups == null) {
+            advertisedGroups = new ArrayList<>();
+        }
+
+        return advertisedGroups;
+    }
+
+    public void setAdvertisedGroups(List<String> advertisedGroups) {
+        this.advertisedGroups = advertisedGroups;
+    }
+
+    /**
+     * The list of individual IP ranges to advertise in ``CUSTOM`` mode.
+     *
+     * @subresource gyro.google.compute.RouterIpRange
+     */
+    @Updatable
+    public List<RouterIpRange> getIpRange() {
+        if (ipRange == null) {
+            ipRange = new ArrayList<>();
+        }
+
+        return ipRange;
+    }
+
+    public void setIpRange(List<RouterIpRange> ipRange) {
+        this.ipRange = ipRange;
+    }
+
+    @Override
+    public String primaryKey() {
+        return getAsn().toString();
+    }
+
+    @Override
+    public void copyFrom(com.google.api.services.compute.model.RouterBgp model) {
+        setAsn(model.getAsn());
+        setAdvertiseMode(model.getAdvertiseMode());
+        setAdvertisedGroups(model.getAdvertisedGroups());
+
+        if (model.getAdvertisedIpRanges() != null) {
+            setIpRange(model.getAdvertisedIpRanges().stream().map(i -> {
+                RouterIpRange routerIpRange = new RouterIpRange();
+                routerIpRange.copyFrom(i);
+                return routerIpRange;
+            }).collect(Collectors.toList()));
+        }
+    }
+
+    com.google.api.services.compute.model.RouterBgp toRouterBgp() {
+        com.google.api.services.compute.model.RouterBgp routerBgp = new com.google.api.services.compute.model.RouterBgp();
+        routerBgp.setAsn(getAsn());
+        routerBgp.setAdvertiseMode(getAdvertiseMode());
+        routerBgp.setAdvertisedGroups(getAdvertisedGroups());
+        routerBgp.setAdvertisedIpRanges(getIpRange().stream()
+            .map(RouterIpRange::toRouterAdvertisedIpRange)
+            .collect(Collectors.toList()));
+
+        return routerBgp;
+    }
+
+    @Override
+    public List<ValidationError> validate(Set<String> configuredFields) {
+        List<ValidationError> errors = new ArrayList<>();
+
+        if (!getAdvertisedGroups().isEmpty() && getAdvertiseMode() != null && !getAdvertiseMode().equals("CUSTOM")) {
```

Comment:
both the validation can be combined as the check for the advertise mode is common